adaptation.lua

```diff
@@ -0,0 +1,128 @@
+-- Original opening comment before lua adaptation
+-- View or set cavern adaptation levels
+-- based on removebadthoughts.rb
+
+-- Rewritten by TBSTeun using OpenAI GPT from adaptation.rb
+
+local help = [====[
+
+adaptation
+==========
+View or set level of cavern adaptation for the selected unit or the whole fort.
+
+Usage::
+
+    adaptation <show|set> <him|all> [value]
+
+The ``value`` must be between 0 and 800,000 (inclusive).
+
+]====]
+
+-- Color constants, values mapped to color_value enum in include/ColorText.h
+local COLOR_RESET = -1
+local COLOR_GREEN = 2
+local COLOR_RED = 4
+local COLOR_YELLOW = 14
+
+local function print_color(color, s)
+    dfhack.color(color)
+    dfhack.print(s)
+    dfhack.color(COLOR_RESET)
+end
+
+local function usage(s)
+    if s ~= nil then
+        print(s)
+    end
+    print("Usage: adaptation <show|set> <him|all> [value]")
+end
+
+local args = {...}
+
+local mode = args[1] or "help"
+local who = args[2]
+local value = args[3]
+
+if mode == "help" then
+    usage()
+elseif mode ~= "show" and mode ~= "set" then
+    usage("Invalid mode '" .. mode .. "': must be either 'show' or 'set'")
+end
+
+if who == nil then
```

Comment:
Here and elsewhere, you usually don't compare against nil in Lua unless you need to differentiate between nil and other firms of falseness. Just `if not who then` is idiomatic.

Reply:
done